src/vs/workbench/services/extensions/electron-browser/localProcessExtensionHost.ts

```diff
@@ -683,38 +683,30 @@ export class LocalProcessExtensionHost implements IExtensionHost {
 		return withNullAsUndefined(this._inspectPort);
 	}
 
-	public terminate(): void {
-		if (this._terminating) {
-			return;
-		}
-		this._terminating = true;
-
-		this._toDispose.dispose();
-
-		if (!this._messageProtocol) {
-			// .start() was not called
-			return;
-		}
-
-		this._messageProtocol.then((protocol) => {
-
-			// Send the extension host a request to terminate itself
-			// (graceful termination)
-			protocol.send(createMessageOfType(MessageType.Terminate));
-
-			protocol.getSocket().dispose();
-
-			protocol.dispose();
-
-			// Give the extension host 10s, after which we will
-			// try to kill the process and release any resources
-			setTimeout(() => this._cleanResources(), 10 * 1000);
+	public terminate(): Promise<void> {
```

Comment:
Are these changes specific to the renderer reuse flag or can they be ported to `main` independently if they are useful? Asking because now the extension host is not spawned from the workbench anymore so I wonder if they have purpose even now already?

Reply:
Those changes were previously there to slow down `onWillShutdown` because otherwise,
1. Adding a breakpoint to that function and stepping through it would result in a crash, because the environment was being destroyed/shut down anyway.
2. If the resources weren't cleaned, a zombie process would occur on macOS.

Now with Alex's changes, point 2 is irrelevant. I believe point 1 still occurs, but considering how one has to add a breakpoint there to even repro the issue, it's unclear to me whether there is any problems with us leaving that code as-is. I have reverted that code, and now the PR only adds a runtime flag.